lib/stages/ReadGain.hpp

```diff
@@ -0,0 +1,102 @@
+/**
+ * @file
+ * @brief read in new gain file for FRB/PSR when available
+ *  - ReadGain : public kotekan::Stage
+ */
+
+#ifndef READ_GAIN
+#define READ_GAIN
+
+#include "Stage.hpp"
+#include "restServer.hpp"
+
+#include <condition_variable>
+#include <mutex>
+#include <vector>
+
+using std::vector;
+
+/**
+ * @class ReadGain
+ * @brief read in new gain files for FRB/PSR when available
+ *
+ * @par Buffers
+ * @buffer gain_frb_buf Array of gains size 2048*2
+ *     @buffer_format Array of @c floats
+ *     @buffer_metadata none
+ * @buffer gain_psr_buf Array of gains size 2048*2*nbeam
+ *     @buffer_format Array of @c floats
+ *     @buffer_metadata none
+ *
+ * @conf   num_elements         Int (default 2048). Number of elements
+ * @conf   num_pulsar           Int (default 10). Number of pulsar beams to be formed
+ * @conf   scaling              Float (default 1.0). Scaling factor on gains
+ * @conf   default_gains        Float array (default 1+1j). Default gain value if gain file is
+ * missing
+ *
+ * The gain path is registered as a subscriber to an updatable config block.
+ *
+ * @author Cherry Ng
+ *
+ */
+
```

Comment:
Remove this space, it might cause issues with doxygen associating the comments to the code block.